File: Java/src/JSJ/files.java
// File: FileUtils.java
package JSJ;
import java.io.*;
import java.nio.file.*;

public class files {

    /**
     * Writes content to a file. Overwrites if the file already exists.
     *
     * @param filePath The file path where content should be written.
     * @param content The content to write to the file.
     * @throws IOException If an IO error occurs during the operation.
     */
    public static void writeToFile(String filePath, String content) throws IOException {
        Files.write(Paths.get(filePath), content.getBytes());
    }

    /**
     * Reads content from a file.
     *
     * @param filePath The file path to read from.
     * @return The content of the file as a string.
     * @throws IOException If an IO error occurs or the file doesn't exist.
     */
    public static String readFromFile(String filePath) throws IOException {
        return Files.readString(Paths.get(filePath));
    }

    /**
     * Appends content to a file. Creates the file if it doesn't exist.
     *
     * @param filePath The file path where content should be appended.
     * @param content The content to append to the file.
     * @throws IOException If an IO error occurs during the operation.
     */
    public static void appendToFile(String filePath, String content) throws IOException {
        Files.write(Paths.get(filePath), content.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
    }

    /**
     * Deletes a file.
     *
     * @param filePath The file path of the file to delete.
     * @return True if the file was successfully deleted, false otherwise.
     */
    public static boolean deleteFile(String filePath) {
        try {
            return Files.deleteIfExists(Paths.get(filePath));
        } catch (IOException e) {
            System.err.println("Error deleting file: " + e.getMessage());
            return false;
        }
    }

    /**
     * Example usage of FileUtils methods.
     */
    public static void main(String[] args) {
        String filePath = "example.txt";

        try {
            // Write to file
            writeToFile(filePath, "Hello, world!\n");

            // Append to file
            appendToFile(filePath, "This is an appended line.\n");

            // Read from file
            String content = readFromFile(filePath);
            System.out.println("File Content:\n" + content);

            // Delete file
            boolean deleted = deleteFile(filePath);
            System.out.println("File deleted: " + deleted);

        } catch (IOException e) {
            System.err.println("An error occurred: " + e.getMessage());
        }
    }
}

File: Rust/JSRS/src/JSRS/files.rs
use std::fs::{self, File};
use std::io::{self, Read, Write};
use std::path::Path;

struct FileManager;

impl FileManager {
    /// Reads the contents of a file and returns it as a String.
    fn read_file(file_path: &str) -> Result<String, io::Error> {
        let mut file = File::open(file_path)?;
        let mut contents = String::new();
        file.read_to_string(&mut contents)?;
        Ok(contents)
    }

    /// Writes the given content to a file, overwriting if the file already exists.
    fn write_file(file_path: &str, content: &str) -> Result<(), io::Error> {
        let mut file = File::create(file_path)?;
        file.write_all(content.as_bytes())?;
        Ok(())
    }

    /// Appends the given content to a file, creating the file if it doesn't exist.
    fn append_to_file(file_path: &str, content: &str) -> Result<(), io::Error> {
        let mut file = OpenOptions::new()
            .create(true)
            .append(true)
            .open(file_path)?;
        file.write_all(content.as_bytes())?;
        Ok(())
    }

    /// Deletes the specified file.
    fn delete_file(file_path: &str) -> Result<(), io::Error> {
        fs::remove_file(file_path)?;
        Ok(())
    }

    /// Checks if a file exists at the given path.
    fn file_exists(file_path: &str) -> bool {
        Path::new(file_path).exists()
    }
}